xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp

```diff
@@ -324,6 +324,9 @@ void CGUIDialogSmartPlaylistEditor::OnType()
     return;
 
   m_playlist.SetType(ConvertType(allowedTypes[newSelected]));
+  // remove any grouping that could be left over when changing the type of playlist to songs
+  if (ConvertType(allowedTypes[newSelected]) == "songs")
+      m_playlist.SetGroup(CSmartPlaylistRule::TranslateGroup(FieldNone));
```

Comment:
Really tempted to make fixing the playlist editor a separate PR rather than lose the GUI change in here (with music stuff where no one ever looks). 

Whenever the current type of playlist contents does not have groupings (so will be greyed), or the current grouping is invalid, it should be set to FieldNone.

Reply:
The easiest way is to just drop the check so that anytime the type is changed, the grouping is reset to none. Separate PR is fine.  Probably I should have written `if (ConvertType(allowedTypes[newSelected]) == TYPE_SONGS)` initially though.  Didn't really dig about too much in there to see what was possible at the time.

Reply:
No, when I talked before about making it "generic" I mean don't put hard coded checks in here for valid groupings that are configured by static data in SmartPlaylist.cpp

Reply:
Like this
```
  // Remove any invalid grouping left over when changing the type
  Field currentGroup = CSmartPlaylistRule::TranslateGroup(m_playlist.GetGroup().c_str());
  if (currentGroup != FieldNone)
  {
    std::vector<Field> groups = CSmartPlaylistRule::GetGroups(m_playlist.GetType());
    if (std::find(groups.begin(), groups.end(), currentGroup) == groups.end())
      m_playlist.SetGroup(CSmartPlaylistRule::TranslateGroup(FieldNone));
  }
```

Reply:
But it belongs in a separate PR not hidden in a music feature

Reply:
So I raised the PR for it (gotta get some Git credit), `CGUIDialogSmartPlaylistEditor` changes can be dropped from this.

Reply:
Totally agree it should be a separate PR.

Is `FieldUnknown` not a better option to use as that shows as `default` in the GUI?  Everything has a default option, but not everything has a `none` option.
Also, do you think it's better to identify whether the grouping exists in the new type of playlist or just to reset the field if the type changes as the user is effectively starting again with their playlist?  Guess that can all be decided later.

Reply:
>Is FieldUnknown not a better option to use as that shows as default in the GUI? Everything has a default option, but not everything has a none option.

Good point. 

>Also, do you think it's better to identify whether the grouping exists in the new type of playlist or just to reset the field if the type changes as the user is effectively starting again with their playlist? 

I think for ease of use should groupigs be common across playlist types that letting valid values stay is the best route rather than reset

>Guess that can all be decided later.

Yes it can, please comment on that PR :)